plugins/c8y_remote_access_plugin/src/auth.rs

```diff
@@ -0,0 +1,24 @@
+use c8y_api::http_proxy::C8yJwtTokenRetriever;
+use c8y_api::http_proxy::C8yMqttJwtTokenRetriever;
+use miette::IntoDiagnostic;
+use tedge_config::TEdgeConfig;
+
+pub struct Jwt(String);
+
+impl Jwt {
```

Comment:
This struct is a bit redundant now as the user could have used the `C8yMqttJwtTokenRetriever` directly without this `Jwt` struct indirection.

Reply:
Regardless of whether I have the struct or not, I'd still want to abstract away retrieving the token and doing the necessary error handling. The struct adds very little noise on top of this IMO.

Reply:
Oh yeah, I overlooked that `miette::Result<Jwt>` conversion aspect.